### PR TITLE
Add global event handler attributes

### LINKED ISSUE #141: Define ongamepadconnected/ongamepaddisconnected attributes
Generally events are paired with corresponding attributes, so we probably want to define them.

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #141 

The following tasks have been completed:

 * [X] Modified Web platform tests  - tests will be generated automatically once merged. 

Implementation commitment:

 * [x] WebKit (https://webkit.org/b/223860)
 * [ ] Chromium (https://crbug.com/1192878)
 * [x] Gecko (https://bugzil.la/1697093)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/gamepad/pull/144.html" title="Last updated on Mar 29, 2021, 2:21 AM UTC (b90c0e7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/gamepad/144/05bad22...b90c0e7.html" title="Last updated on Mar 29, 2021, 2:21 AM UTC (b90c0e7)">Diff</a>

### CODE REVIEW

**Review by marcoscaceres — APPROVED** [review]
Sorry for the delay @saschanaz. The IDL and prose looks good. 

@saschanaz, will you add these to Gecko or would you prefer I did it?

**Comment by marcoscaceres** [comment]
Filed Gecko [1697093](https://bugzilla.mozilla.org/show_bug.cgi?id=1697093) and  [1697094](https://bugzilla.mozilla.org/show_bug.cgi?id=1697094).

**Comment by saschanaz** [comment]
Hmm, I guess I can do that. Thanks for filing them 👍

File: index.html
Comment (saschanaz) [inline index.html:790]: Should this be `partial interface Window` instead, because we don't really want to fire them on `document.body`? @marcoscaceres

Comment (marcoscaceres) [reply index.html:790]: Yes, I think you are right.

Comment (saschanaz) [reply index.html:790]: Checked some precedents and it seems anything using `partial interface Window` for adding event handlers are deprecated. (from https://w3c.github.io/deviceorientation/#idl-index, WebVR and some obsolete specs). Doing `WindowEventHandlers` also allows `<body ongamepadconnected="">` which is probably a good thing.

Comment (saschanaz) [reply index.html:790]: BTW I submitted a patch, is any other thing blocking merging here?

Comment (marcoscaceres) [reply index.html:790]: No blockers - just need ok from @nondebug or @JamesHollyer to confirm this matches Chrome.

Comment (marcoscaceres) [reply index.html:790]: Also, could you kindly check what the merge conflict is?

Comment (nondebug) [reply index.html:790]: This looks fine to me. Chrome strangely has no IDL for the Gamepad event handlers, but this matches the IDL for other Window events.

Comment (marcoscaceres) [reply index.html:790]: @nondebug, you are ok to add this to Chrome, right? Would you like me to file a bug for it?

Comment (nondebug) [reply index.html:790]: Filed https://crbug.com/1192878 to fix this in Chrome

**Comment by marcoscaceres** [comment]
Filed WK bug https://bugs.webkit.org/show_bug.cgi?id=223860

**Comment by marcoscaceres** [comment]
@saschanaz, the Editors are meeting next week, so will merge then. Hope that ok!

**Comment by saschanaz** [comment]
@marcoscaceres Did the editors agree on this? 👀

**Comment by marcoscaceres** [comment]
Meeting is on Friday, but I don't see any blockers. I expect we will merge then.

**Review by nondebug — APPROVED** [review]

**Comment by marcoscaceres** [comment]
thanks again @saschanaz!

**Pull request merged**
